src/compile/data/bin.ts

```diff
@@ -18,37 +18,41 @@ function numberFormatExpr(expr: string, format: string) {
 
 function parse(model: Model): Dict<VgTransform[]> {
   return model.reduceFieldDef(function(binComponent: Dict<VgTransform[]>, fieldDef: FieldDef, channel: Channel) {
-    const bin = model.fieldDef(channel).bin;
+    const fieldDefBin = model.fieldDef(channel).bin;
+    const bin: Bin = isBoolean(fieldDefBin) ? {} : fieldDefBin;
+    if (bin && !(<VgTransform>bin).maxbins && !(<VgTransform>bin).steps) {
```

Comment:
1) You have to check if bin is defined or not first.  If not there is no point processing the rest. 

2) These casting `<VgTransform>` seem totally out of place.